src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java

```diff
@@ -220,17 +220,22 @@ public Case resolveUserRef(Case useCase) {
         useCase.getUsers().clear();
         useCase.getNegativeViewUsers().clear();
         useCase.getUserRefs().forEach((id, permission) -> {
-            List<String> userIds = getExistingUsers((UserListFieldValue) useCase.getDataSet().get(id).getValue());
-            if (userIds != null && userIds.size() != 0 && permission.containsKey("view") && !permission.get("view")) {
-                useCase.getNegativeViewUsers().addAll(userIds);
-            } else if (userIds != null && userIds.size() != 0) {
-                useCase.addUsers(new HashSet<>(userIds), permission);
-            }
+            resolveUserRefPermissions(useCase, id, permission);
         });
         useCase.resolveViewUsers();
+        taskService.resolveUserRef(useCase);
         return repository.save(useCase);
     }
 
+    private void resolveUserRefPermissions(Case useCase, String userListId, Map<String, Boolean> permission) {
+        List<String> userIds = getExistingUsers((UserListFieldValue) useCase.getDataSet().get(userListId).getValue());
+        if (userIds != null && userIds.size() != 0 && permission.containsKey("view") && !permission.get("view")) {
+            useCase.getNegativeViewUsers().addAll(userIds);
+        } else if (userIds != null && userIds.size() != 0) {
```

Comment:
this is the same condition as part of the first `if` branch, please merge and refactor `if-else`

Reply:
Done